pkg/issuer/acme/dns/dns.go

```diff
@@ -253,15 +253,22 @@ func (s *Solver) solverForChallenge(ctx context.Context, issuer v1alpha2.Generic
 		}
 	case providerConfig.Cloudflare != nil:
 		dbg.Info("preparing to create Cloudflare provider")
-		apiKeySecret, err := s.secretLister.Secrets(resourceNamespace).Get(providerConfig.Cloudflare.APIKey.Name)
-		if err != nil {
-			return nil, nil, fmt.Errorf("error getting cloudflare service account: %s", err)
+		apiKeySecret, keyErr := s.secretLister.Secrets(resourceNamespace).Get(providerConfig.Cloudflare.APIKey.Name)
```

Comment:
These should be different error checks.

We need to determine which auth strategy we are using. If none or both are set we should error.

You can then get the corresponding secret to GET based on that and error check solely on that.